### ISSUE TITLE
Add User Managed Access authorization support

### ISSUE THREAD

**Issue by sberyozkin** [quarkusio/quarkus#4817]
**Description**
Please see https://kantarainitiative.org/confluence/display/uma/Home.

**Implementation ideas**
Keycloak already supports it, the only 'restriction' is that the adapter code needs to be generic for it to work with any IDPs implementing UMA.

**Comment by sberyozkin** [comment]
Closing it until the users will start asking about UMA

**Comment by marc-christian-schulze** [comment]
Looking forward to have UMA supported in Quarkus so we can enable resource servers to dynamically create resources and manage permissions like shown in the Keycloak quickstart example:
https://github.com/keycloak/keycloak-quickstarts/tree/latest/app-authz-uma-photoz

**Comment by carlosthe19916** [comment]
Trying to reproduce the Keycloak's example "app-authz-uma-photoz" using Quarkus I realized that the creation of "Resources owned by a user" is not supported yet. I'd like to apply UMA to have control of resources in this kind of context:

- There are several Bank accounts
- Every account is owned only by a specific user
- A bank account owner can share his own resources with other users.

This is a general example but this is the kind of context that Keycloak - UMA (authorization services) claim to be for. It would be great to have Keycloak-UMA supported in Quarkus too.

**Comment by sberyozkin** [comment]
I'd like to close this issue for now as it is unlikely we will be able to prioritize on it in the short term - but can re-open it if there will be more user queries

**Comment by kdubb** [comment]
@sberyozkin Any chance we can resurrect this? I'm willing to pitch in as we'd like to use UMA in one of our Quarkus services.

**Comment by kdubb** [comment]
@sberyozkin FYI, like @carlosthe19916 I was following the `app-authz-uma-photoz` example. In my case I seem to be able to create "Resources owned by a user" just fine using an injected `AuthzClient`. 

My issue is that when requesting the `grant_type=urn:ietf:params:oauth:grant-type:uma-ticket` token it returns an `access_denied` error (with description `not_authorized`).

**Comment by sberyozkin** [comment]
Let me re-open this issue again.
@kdubb Perhaps it can make sense to extend `quarkus-oidc` with `quarkus-uma`.

By the way, I can enhance `OidcClient` to support requesting such grant tokens

**Comment by sberyozkin** [comment]
Hi @kdubb Are you still interested in the `quarkus-uma` idea ? Do you see any demands for this feature in production ?

**Comment by kdubb** [comment]
We abandoned the idea for a couple reasons.

1. Support
   I had hacked some code to get it "working" with Keycloack but it was far from a usable state.

2. Complexity
   The spec and the implementation seem overly complex and the setup in Keycloak was downright confusing (seemingly the spec's fault).

3. Zanzibar
   It's simple for simple things and pretty simple for complex things. Additionally, I created the original extension for testing in less than a day.
